File: Problemas/Codeforces/2123/C/C.py
# https://codeforces.com/problemset/problem/2123/C

def presufix():
    nTestes = int(input())
    for _ in range(nTestes):
        tamanho = int(input())
        lista = [int(x) for x in input().split()]

        pre = [lista[0]]
        suf = [0] * tamanho
        suf[-1] = lista[-1]

        for i in range(1, tamanho):
            pre.append(min(pre[-1], lista[i]))
        for i in range(tamanho - 2, -1, -1):
            suf[i] = max(suf[i + 1], lista[i])

        print("".join("1" if lista[i] in (pre[i], suf[i]) else "0" for i in range(tamanho)))

presufix()

File: Problemas/Codeforces/2123/C/C.cpp
// https://codeforces.com/problemset/problem/2123/C

#include <bits/stdc++.h>

using namespace std;

#define INF 1ll<<60

typedef long long ll;

void solve()
{
    ll n, i;
    cin >> n;
    vector<ll> a(n + 1), pre(n + 2, INF), suf(n + 2, 0);
    
    for (i = 1; i <= n; i++)
    {
        cin >> a[i];
        pre[i] = min(pre[i - 1], a[i]);
    }
    
    for (i = n; i >= 1; i--)
    {
        suf[i] = max(suf[i + 1], a[i]);
    }
    
    for (i = 1; i <= n; i++)
    {
        if (a[i] == pre[i] or a[i] == suf[i])
            cout << 1;
        else
            cout << 0;
    }
    cout << endl;
}

int main()
{
    ll t;
    cin >> t;
    while (t--) solve();
}

